src/Parallel/ResourceInfo.hpp

```diff
@@ -84,4 +91,91 @@ struct SingletonInfoHolder {
   std::optional<size_t> proc_{std::nullopt};
   bool exclusive_{false};
 };
+
+template <typename ParallelComponents>
+struct SingletonPack;
+
+/// \cond
+// LCOV_EXCL_START
+// Special case needed when the parameter pack passed to SingletonPack is empty.
+// This is only necessary to get things to compile and shouldn't be used.
```

Comment:
If it shouldn't be used, should there be `ERROR`s in the functions so that things can't be used? Another way of getting around this is to leave this case undefined, and do `tmpl::list<ParallelComponent0, ParallelComponents...>` in the general implementation. It'll then only match if there's at least one component

Reply:
The way I have things set up for a future PR, I need the empty list case to be defined and `pup`able, it just won't ever be used. (And by "used" I mean the `get` function will never be called.) I'll add an `ERROR` to the `get` function.

Reply:
Okay, you sohuld then test the class specialization and remove the `LCOV_EXCL_START` (and stop)